libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboard.tsx

```diff
@@ -237,9 +237,15 @@ export class ModelAssessmentDashboard extends CohortBasedComponent<
     let newCohorts = [...this.state.cohorts];
     newCohorts[editIndex] = newErrorCohort;
     newCohorts = newCohorts.filter((cohort) => !cohort.isTemporary);
-    this.setState({
-      cohorts: newCohorts
-    });
+    if (switchNew) {
+      const newlySelectedCohort = newCohorts[editIndex];
+      this.setState({
+        cohorts: newCohorts,
+        selectedCohort: newlySelectedCohort
+      });
+      return;
```

Comment:
why not just use an else instead of early return here?  Is this actually preferred in terms of style?
I wonder if you can define the args and pass to this.setState:

```
    let cohortArgs = { cohorts: newCohorts }
    if (switchNew) {
      const newlySelectedCohort = newCohorts[editIndex];
      cohortArgs[selectedCohort] = newlySelectedCohort
    }
    this.setState(cohortArgs);
```